### PR TITLE
Bugfix: broken promise chain within resolve testing tools

### PR DESCRIPTION
**Pull request by max-vasin**

### CODE REVIEW

File: packages/core/resolve-testing-tools/src/aggregate-assertions.ts [hunk 1]
```diff
@@ -19,6 +19,19 @@ type BDDAggregateAssertionContext = {
   [symbol]: BDDAggregateAssertionState
 }
 
+const stringifyError = (error: any): string => {
```
Comment (IhostVlad) [inline packages/core/resolve-testing-tools/src/aggregate-assertions.ts:22]: Does such function *really* considered? Which use cases have forced such counter-intuitive implementation?

| Input | Output |
|-------|--------|
| 0     | "no error"      |
| 1     | "1"      |
| false     | "no error"      |
| true     | "true"      |
| ""     | "no error"      |
| "abc"    | "abc"      |
| new String("abc")    | "\\"abc\\""      |
| String("abc")    | "abc"      |
| new Error()    | "\\"\\""      |
| new Error("abc")    | "abc"      |

File: packages/core/resolve-testing-tools/src/aggregate-assertions.ts [hunk 2]
```diff
@@ -19,6 +19,19 @@ type BDDAggregateAssertionContext = {
...
+  if (!error) {
```
Comment (IhostVlad) [inline packages/core/resolve-testing-tools/src/aggregate-assertions.ts:23]: `error != null`

File: packages/core/resolve-testing-tools/src/aggregate-assertions.ts [hunk 3]
```diff
@@ -19,6 +19,19 @@ type BDDAggregateAssertionContext = {
...
+    return 'no error'
+  }
+  if (typeof error === 'string') {
```
Comment (IhostVlad) [inline packages/core/resolve-testing-tools/src/aggregate-assertions.ts:26]: `error.constructor === String`

File: packages/core/resolve-testing-tools/src/aggregate-assertions.ts [hunk 4]
```diff
@@ -19,6 +19,19 @@ type BDDAggregateAssertionContext = {
...
+    return error
+  }
+  if (error.message) {
```
Comment (IhostVlad) [inline packages/core/resolve-testing-tools/src/aggregate-assertions.ts:29]: `error instanceof Error && error.message != null && error.message.constructor === String`

File: packages/core/resolve-testing-tools/src/aggregate-assertions.ts (hunk 1 above)
Comment (timbset) [reply packages/core/resolve-testing-tools/src/aggregate-assertions.ts:22]:
```js
const stringifyError = (error: any): string => {
  if (error == null) {
    return 'no error'
  }
  return error.toString()
}
```
Maybe like this?

**Pull request merged**